src/time/duration.rs

```diff
@@ -70,6 +70,34 @@ impl Duration {
     pub fn from_secs_f32(secs: f32) -> Duration {
         std::time::Duration::from_secs_f32(secs).into()
     }
+
+    /// Returns the number of whole seconds contained by this `Duration`.
+    #[must_use]
+    #[inline]
+    pub const fn as_secs(&self) -> u64 {
+        self.0 / 1_000_000_000
+    }
+
+    /// Returns the number of whole microseconds contained by this `Duration`.
+    #[must_use]
+    #[inline]
+    pub const fn as_micros(&self) -> u128 {
+        (self.0 / 1_000_000) as u128
```

Comment:
Millis and micros are swapped here.

This reminded me to go back and double check the other direction, and I got this wrong as well, should be from_nanos. I’d fix it but I’m on mobile. https://github.com/yoshuawuyts/wstd/blob/5a756c8c4200f2083b3d4b6e9916d91a4bde8347/src/time/instant.rs#L33

Reply:
I've now fixed both, and added a test.